api/decode.py

```diff
@@ -0,0 +1,196 @@
+from flask import Flask, abort, json, jsonify
+import hashlib
+import pybitcointools
+from decimal import *
+from rpcclient import *
+
+app = Flask(__name__)
+app.debug = True
+
+@app.route('/<rawhex>')
+def decode_handler(rawhex):
+  return jsonify(decode(rawhex))
+
+
+def decode(rawhex):
+
+  transaction = decoderawtransaction(rawhex)['result']
+  rawtx = getrawtransaction(transaction['vin'][0]['txid'])
+  senders  = rawtx['result']['vout'][transaction['vin'][0]['vout']]['scriptPubKey']['addresses']
```

Comment:
Care must be taken here: the sender is selected based on the combined values of all inputs, e.g.
Vin 0: sourceA, 0.5 BTC
Vin 1: sourceB, 0.4 BTC
Vin 2: sourceB, 0.2 BTC
... then the sender is sourceB.

But even more unfortunally: any other script type other than pay-to-pubkey-hash and pay-to-script-hash invalidates the whole transaction, if it's an input.

Related:
https://github.com/mastercoin-MSC/spec/issues/224 "Feature request: relax input and output requirements, P2SH"
https://github.com/mastercoin-MSC/spec/issues/225 "Easier transaction input selection"